location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,466 @@
+                feature_dict.get('rms').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('spectral_centroid').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('spectral_bandwidth').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('spectral_contrast').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('spectral_flatness').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('spectral_rolloff').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('poly_features').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('tonnetz').tolist())]
+                                + [delete_outer_list(
+                feature_dict.get('zero_crossing_rate').tolist())])
+
+
+def label_selection(labels_list, labels_set):
+    """Returns a 1 if any label in the labels_list is in labels_set else a 0.
+
+    Iterates through the labels_list and returns a 1 if any label is in the
+    labels_set, and returns a 0 otherwise.
+
+    Args:
+        labels_list: a list of labels in the format found in the audioset csv
+        labels_set: a set of labels past in via command line arguments in the
+                    format found in the audioset csv
+
+    Returns:
+        A integer, a 0 or 1, depending on if any label in labels_list is present
+        in labels_set
+    """
+    translated_labels = []
+    for label in labels_list:
+        translated_labels.append(label_id_search(FLAGS.src_dir, label))
+    for label in translated_labels:
+        if label in labels_set:
+            return 1
+        else:
+            return 0
+
+
+def label_list_to_set(translated_labels_list):
+    """Converts a list of labels to a set of labels.
+
+    Converts a list of labels in the format found in the audioset csv to a set
+    of labels, then returns that set.
+
+    Args:
+        translated_labels_list: a list of labels in the format found in the
+            audioset csv passed in as command line arguments
+
+    Returns:
+        A set of labels in the format found in the audioset csv
+    """
+    labels_set = set(translated_labels_list)
+    return labels_set
+
+
+def delete_outer_list(alist):
+    """Removes outer lists from a nested list.
+
+    Removes outer lists from a nested list until the list has more than one
+    element.
+
+    Args:
+        alist: Inputted list from which to remove any unnecessary outer lists.
+
+    Returns:
+        A list with all unnecessary outer lists removed. For example:
+
+        [[1,2,3,4],[1,3,5,2]] or [1,5,3,2,6,9]
+    """
+    if len(alist) == 1 and isinstance(alist, list):
+        return delete_outer_list(alist[0])
+    return alist
+
+
+def output_csv_linear():
```

Comment:
not sure what "_linear" means here

Reply:
I meant is as the default way of processing the data (no multithreading or multiprocessing). However, I can change it to just be output_csv().